configure.ac

```diff
@@ -604,6 +604,37 @@ AM_CONDITIONAL([OPENTHREAD_ENABLE_CERT_LOG], [test "${enable_cert_log}" = "yes"]
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_CERT_LOG],[${OPENTHREAD_ENABLE_CERT_LOG}],[Define to 1 if you want to enable log for certification test])
 
 #
+# Dhcp Service
+#
+
+AC_ARG_ENABLE(dhcp,
+    [AS_HELP_STRING([--enable-dhcp],[Enable dhcp support @<:@default=no@:>@.])],
+    [
+        case "${enableval}" in
+
+        no|yes)
+            enable_dhcp=${enableval}
+            ;;
+
+        *)
+            AC_MSG_ERROR([Invalid value ${enable_dhcp} for --enable-dhcp])
+            ;;
+        esac
+    ],
+    [enable_dhcp=no])
+
+if test "$enable_dhcp" = "yes"; then
+    OPENTHREAD_ENABLE_DHCP=1
```

Comment:
For example, in [libopenthread_k.vcxproj](https://github.com/openthread/openthread/blob/master/etc/visual-studio/libopenthread_k.vcxproj) you can see the `PreprocessorDefinitions` section defines `MBEDTLS_CONFIG_FILE="mbedtls-config.h"`. That is the same as if you had the following at the top of every file:

```
#define MBEDTLS_CONFIG_FILE "mbedtls-config.h"
```